---
title: "Kolokvium"
author: "Peter Manik"
date: "30/05/2022"
output: html_document
editor_options: 
  markdown: 
    wrap: 72
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

```

# Introduction

![](C:/Users/peter/AppData/Local/RStudio/tmp/paste-DC7E7C4B.png)

BirA was fused with the invadopodia-specific protein **Tks5α** and
mutant which lacked the membrane-targeting PX domain (**Tks5β**). After
addition of biotin, both fusion proteins biotinylated proteins in their
vicinity. Biotinylated proteins were isolated by affinity capture and
identified by mass spectrometry.

# Load libraries

-   here (path definitions)
-   DEP (proteomics workflow)
-   dplyr (basic data operations)
-   SummarizedExperiment (normalization purposes)
-   DT (output table generation)
-   ggplot2 (correlation plot)
-   gprofiler2; enrichR (GSEA)

```{r libraries, message=FALSE, warning=FALSE}
library(ggplot2)
library(ggrepel)
library(enrichR)
library(gprofiler2)
library(here)
library(DEP)
library(dplyr)
library(DT)
library(SummarizedExperiment)
```

# Data input

As input, we start with **kolokvium_PXD015847_proteinGroups.txt** file,
generated by MaxQuant software. We load it as dataframe and look at its
parameters.

```{r data-input, message=FALSE, warning=FALSE}

data <- read.delim(here("data","kolokvium_PXD015847_proteinGroups.txt"))
dim(data)
```

# Data preparation

## Contaminants filtering

We will filter out the most common contaminants:

-   Reverse sequences
-   Only identified by site proteins
-   cRAP protein sequences (e.g common laboratory proteins, proteins
    added accidentally by dust/physical contact, etc.)
-   keratins

By change of nrow() value you can see how many proteins were filtered
out.

```{r contaminants-filtering, message=FALSE, warning=FALSE}
data <- data %>%
  filter(Reverse != "+") %>% 
  filter(!grepl("cRAP", Majority.protein.IDs)) %>% 
  filter(Only.identified.by.site != "+") %>% 
  filter(!grepl("keratin", Fasta.headers)) %>%
  filter(!grepl("Keratin", Fasta.headers)) %>% 
  filter(!grepl("PE=5", Fasta.headers))

nrow(data)
```

1158 proteins identified before filtering -> 1097 proteins after
filtering

## Unique identifiers

To evaluate abundance of proteins, each protein must have unique
identifier (protein name/ ID)

```{r unique-identifiers-a, message=FALSE, warning=FALSE}
# Are there any duplicated gene names?
data$Gene.names %>% duplicated() %>% any()

# Make a table of duplicated gene names
data %>% group_by(Gene.names) %>% summarize(frequency = n()) %>% 
  arrange(desc(frequency)) %>% filter(frequency > 1)
```

14 proteins are without name we should assign protein_id as their name.

We create new column "name" containing protein names or protein_id if
protein does not have a name.

Now we have 0 proteins without a name :)

```{r unique-identifiers-b, message=FALSE, warning=FALSE}
# Make unique names using the annotation in the "Gene.names" column as primary names and the annotation in "Protein.IDs" as name for those that do not have an gene name.
data_unique <- make_unique(data, "Gene.names", "Protein.IDs", delim = ";")
data_unique %>% group_by(name) %>% summarize(frequency = n()) %>% 
  arrange(desc(frequency)) %>% filter(frequency > 1)
```

## SummarizedExperiment

Dataframe is transformed to SummarizedExperiment object for downstream
analyses.

LFQ was chosen for calculation of protein abundance. Names of samples
and conditions were extracted from colnames using gsub + regex

```{r summarized-experiment-CK1a-1-excluded, message=FALSE, warning=FALSE}

intensity_columns <- grep("LFQ.", colnames(data_unique)) 

# create experimental design
label_columns <- grep("Peptides.", colnames(data_unique)) 
label_row_names <- colnames(data_unique)[label_columns]
labels <- gsub("^Peptides.","",label_row_names)
condition <- gsub("_[a-c]$","",gsub("^[0-9]_","",labels))
exp_design <- data.frame(
  label = labels,
  condition = condition,
  replicate = c(rep(1:3, times = 3),rep(4:6, times = 3),c(1:3))
)

# Define the variable types
exp_design$label <- as.character(exp_design$label)
exp_design$condition <- as.character(exp_design$condition)
exp_design$replicate <- as.numeric(exp_design$replicate)

#create SummarizedExperiment object
data_se <- make_se(data_unique, intensity_columns, exp_design)
```

```{r SE-intermezzo, message=FALSE, warning=FALSE, include = FALSE}
# plot identifications per each condition
data_se
dim(data_se)

# dimnames(data_se)
# colData(data_se)
# assay(data_se)
head(assay(data_se))

#data_se$condition
#data_se@assays@data@listData
```

## Number of proteins identified in each sample

We expect similar number of proteins identified in condition between
replicates. NT does not contain Biotin ligase- I would expect less
proteins identified there.

```{r plot-numbers, message=FALSE, warning=FALSE}
# plot identifications per each condition
plot_numbers(data_se)
```

## Data filtering

Firstly, we plot a barplot of the protein identification overlap between
samples. Most of the proteins were present in all of the samples.

```{r filter-data-a, message=FALSE, warning=FALSE}
plot_frequency(data_se)
```

Mild filtering parameters - protein is counted if it was identified in
at least 2 replicates.

```{r filter-data-b, message=FALSE, warning=FALSE}
data_filt <- filter_missval(data_se, thr = 1)
plot_numbers(data_filt)
plot_frequency(data_filt)
```

# Normalization

**LoessF normalization**

```{r normalization-loessF, message=FALSE, warning=FALSE}
data_norm_loess <- data_filt
assay(data_norm_loess)<-limma::normalizeCyclicLoess(assay(data_norm_loess))
meanSdPlot(data_norm_loess)
plot_normalization(data_filt, data_norm_loess)
```

## Imputation of NAs

BioID data often contain lots of missing values in control, these need
to be imputed for downstream statistical analysis. Since 0s in this
context represent proteins not biotinylated in Neg CTRL it is type MNAR.

-   **MNAR**: proteins below detection limit, low abundance proteins

Various methods available for MNAR imputation I chose MinProb

-   MinProb: imputation by random draws from a Gaussian distribution
    centered to a minimal value (q-th quantile (5% quantile used)

Heatmap of missing proteins in at least 1 condition (white = NA; black =
present)

```{r imputation-heatmap, message=FALSE, warning=FALSE}
plot_missval(data_norm_loess)
```

Plot intensity distributions and cumulative fraction of proteins with
and without missing values

```{r imputation-distribution, message=FALSE, warning=FALSE}
plot_detect(data_norm_loess)
```

Impute the missing values

```{r imputation, message=FALSE, warning=FALSE}
data_imp <- impute(data_norm_loess,fun = "MinProb", q = 0.05)
plot_imputation(data_norm_loess, data_imp)

```

Plot PCA for 500 most variable proteins on Y axis variation between
biological replicates can be observed (remember 1-3 are technical
replicates so 1-3 = 1.BR, 4-6 = 2.BR). X axis represent biological
variance - variance caused by specific biotinylation by fusion proteins
(NT, BirA, TKS_WT, TKS_del)

```{r imputation-pca, message=FALSE, warning=FALSE}
plot_pca(data_imp, x = 1, y = 2, n = 500, point_size = 4)
```

# Differential expression

Only BirA used as control in this step because: a) NT is similar to BirA
according to PCA b) it is more suitable control (identifies
non-specifically biotinylated proteins). Alternatively I could merge
BirA with NT and used that as a control. Difference between WT and MUT
version of TKS5 was also tested.

```{r limma, message=FALSE, warning=FALSE}
# Test every sample versus control
data_diff <- test_diff(data_imp, type = "control", control = "BirA")

# Test manually defined comparisons
data_diff_manual <- test_diff(data_imp, type = "manual",
                             test = "BirA_TKS5WT_vs_BirA_TKS5DelPX")
```

We need to **correct for multiple testing** and **set cutoffs** for
calling a protein differentially expressed:

\- logFC -> 1

\- adjusted p-value -> 0.05

```{r limma-cutoff, message=FALSE, warning=FALSE}
dep <- add_rejections(data_diff, alpha = 0.05, lfc = log2(1))
dep_manual <- add_rejections(data_diff_manual, alpha = 0.05, lfc = log2(1))
```

# Data visualization

Correlation of significantly enriched proteins between samples.
Technical replicates are identical, biological are highly similar. As
expected both proteins have similar sets of identified proteins in
comparison to control samples (NT; CTRL).

```{r correlation-matrix, message=FALSE, warning=FALSE}
# Plot the Pearson correlation matrix
plot_cor(dep, significant = TRUE, lower = 0, upper = 1, pal = "Blues")
```

## GSEA

WT vs BirA - I expect enrichment of cytoskeleton/ plasma membrane in
<GO:CC>.

Indeed these categories are one of the top hits.

```{r gsea, message=FALSE, warning=FALSE}
data_results <- get_results(dep)
gostplot(gost(query = c(data_results$name[data_results$BirA_TKS5WT_vs_BirA_significant & data_results$BirA_TKS5WT_vs_BirA_ratio > 0])), capped = F, interactive= T)
```

```{r heatmap-proteins, message=FALSE, warning=FALSE, include=FALSE}
# Plot a heatmap of all significant proteins with the data centered per protein
plot_heatmap(dep, type = "centered", kmeans = TRUE, 
             k = 6, col_limit = 4, show_row_names = FALSE,
             indicate = c("condition", "replicate"))
```

```{r heatmap-contrasts, message=FALSE, warning=FALSE, include=FALSE}
# Plot a heatmap of all significant proteins (rows) and the tested contrasts (columns)
plot_heatmap(dep, type = "contrast", kmeans = TRUE, 
             k = 6, col_limit = 10, show_row_names = FALSE)
```

## Volcano plots

BirA_TKS5WT_vs_BirA_TKS5DelPX

Several interacting partners are lost after deletion of specific domain
from TKS5, including FGD1, RNT4, MAP4 and EIF4A3.

```{r volcano-ubi1, message=FALSE, warning=FALSE}
plot_volcano(dep_manual, contrast = "BirA_TKS5WT_vs_BirA_TKS5DelPX", label_size = 2, add_names = TRUE, adjusted = TRUE)
```

<!-- BirA_TKS5DelPX_vs_BirA -->

<!-- ```{r volcano-ubi1, message=FALSE, warning=FALSE} -->

<!-- plot_volcano(dep, contrast = "BirA_TKS5DelPX_vs_BirA", label_size = 2, add_names = TRUE, adjusted = TRUE) -->

<!-- ``` -->

<!-- BirA_TKS5WT_vs_BirA -->

<!-- ```{r volcano-ubi4, message=FALSE, warning=FALSE} -->

<!-- plot_volcano(dep, contrast = "BirA_TKS5WT_vs_BirA", label_size = 2, add_names = TRUE, adjusted = TRUE) -->

<!-- ``` -->

<!-- NT_vs_BirA -->

<!-- # ```{r volcano-ubi6, message=FALSE, warning=FALSE} -->

<!-- # plot_volcano(dep, contrast = "NT_vs_BirA", label_size = 2, add_names = TRUE, adjusted = TRUE) -->

<!-- # ``` -->

## Barplots of single proteins

These 2 proteins are enriched only in WT TKS5 but not in mutated form:

```{r barplots-single, message=FALSE, warning=FALSE}
plot_single(dep, proteins = c("FGD1","MAP4"), type = "centered")
```

FGD1 and TKS5 have known function in the formaiton of invadopodia:

![](C:/Users/peter/AppData/Local/RStudio/tmp/paste-6A8CFB9D.png)

<!-- # #Correlation plot of logFC WT_BirA vs DEL_BirA -->

<!-- # ```{r} -->

<!-- # data_results <- get_results(dep) -->

<!-- # coef(lm(BirA_TKS5DelPX_vs_BirA_ratio ~BirA_TKS5WT_vs_BirA_ratio , data = data_results)) -->

<!-- # data_results %>% -->

<!-- #   ggplot(aes(x=BirA_TKS5WT_vs_BirA_ratio, -->

<!-- #              y= BirA_TKS5DelPX_vs_BirA_ratio, -->

<!-- #              label = name -->

<!-- #              # col = BirA_TKS5WT_vs_BirA_significant, -->

<!-- #              # shape = BirA_TKS5DelPX_vs_BirA_significant -->

<!-- #              ))+ -->

<!-- #   geom_point()+ -->

<!-- #   geom_abline(intercept = 0.01397829, slope = 0.86765486) + -->

<!-- #   geom_text_repel(max.overlaps = 20)+ -->

<!-- #   labs(title = 'Correlation plot of WT_BirA vs Del_BirA', -->

<!-- #        x = 'BirA_TKS5WT_vs_BirA' , -->

<!-- #        y = 'BirA_TKS5DelPX_vs_BirA')+ -->

<!-- #   theme_minimal() -->

<!-- # gostres <- gost(query = c(data_results$name[!(data_results$BirA_TKS5DelPX_vs_BirA_significant & data_results$BirA_TKS5DelPX_vs_BirA_ratio > 0) & (data_results$BirA_TKS5WT_vs_BirA_significant & data_results$BirA_TKS5WT_vs_BirA_ratio > 0)]), organism = "hsapiens") -->

<!-- # gostplot(gostres, capped = FALSE, interactive= T) -->

<!-- # data_results_manual <- get_results(dep_manual) -->

<!-- # gostres_man <- gost(query = c(data_results_manual$name[data_results_manual$BirA_TKS5WT_vs_BirA_TKS5DelPX_significant & data_results_manual$BirA_TKS5WT_vs_BirA_TKS5DelPX_ratio>0]), organism = "hsapiens") -->

<!-- # gostplot(gostres_man, capped = FALSE, interactive= T) -->

# SessionInfo()

```{r session-info, message=FALSE, warning=FALSE}
sessionInfo()
```
